src/tests/slot-composer-test.ts

```diff
@@ -73,168 +84,95 @@ recipe
-    // slotComposer = slotComposer
-    //   .expectRenderSlot('A', 'root', {'contentTypes': ['model']})
-    //   .expectRenderSlot('B', 'item', {'contentTypes': ['model']})
-    //   .expectRenderSlot('C', 'item', {'contentTypes': ['model']})
-    //   .expectRenderSlot('B', 'item', {'contentTypes': ['model']})
-    //   .expectRenderSlot('C', 'item', {'contentTypes': ['model']});
-
-    // TODO(sjmiles): context is deprecated
-    //assert.lengthOf(slotComposer.getAvailableContexts(), 1);
+    let {arc, observer, plan} = await initSlotComposer(manifestStr);
 
     plan = plan.clone();
     plan.normalize();
     assert.isTrue(plan.isResolved());
-    await arc.instantiate(plan);
 
-    // TODO(sjmiles): uses various deprecated systems, e.g. `renderSlot` is deprecated
-
-    // assert.deepEqual(['A'], startRenderParticles);
-
-    // const [particleA, particleB, particleC] = arc.activeRecipe.particles;
-    // const rootSlot = slotComposer.getSlotConsumer(particleA, 'root');
-    // const itemSlotId = slotComposer.findContextsByName('item')[0].id;
-    // rootSlot.getInnerContainer = (slotId) => slotId === itemSlotId
-    //     ? {'id1': 'dummy-inner-container-1', 'id2': 'dummy-inner-container-2'}
-    //     : null;
-    // startRenderParticles.length = 0;
-    // await slotComposer.renderSlot(particleA, 'root', {model: {'foo': 'bar'}});
-    // assert.deepEqual(['B', 'C'], startRenderParticles);
-
-    // const gatherRenderings = slotContext => {
-    //   const result = {};
-    //   for (const consumer of slotContext.slotConsumers) {
-    //     for (const [subId, content] of consumer.renderings) {
-    //       if (!result[subId]) result[subId] = [];
-    //       if (content.model) result[subId].push(content.model.title);
-    //     }
-    //   }
-    //   return result;
-    // };
-
-    // const itemSlotContext = slotComposer.findContextsByName('item')[0];
-
-    // await slotComposer.renderSlot(particleB, 'item', {model: {subId: 'id1', title: 'Rendered by B'}});
-    // await slotComposer.renderSlot(particleC, 'item', {model: {subId: 'id2', title: 'Rendered by C'}});
-    // assert.deepEqual({'id1': ['Rendered by B'], 'id2': ['Rendered by C']}, gatherRenderings(itemSlotContext));
-
-    // await slotComposer.renderSlot(particleB, 'item', {model: {subId: 'id2', title: 'B moved to id2'}});
-    // assert.deepEqual({'id1': [], 'id2': ['B moved to id2', 'Rendered by C']}, gatherRenderings(itemSlotContext));
-
-    // await slotComposer.renderSlot(particleC, 'item', {model: {subId: 'id1', title: 'C moved to id1'}});
-    // assert.deepEqual({'id1': ['C moved to id1'], 'id2': ['B moved to id2']}, gatherRenderings(itemSlotContext));
-
-    // TODO(sjmiles): uses old render data, will be repaired in subsequent PR
-    // await slotComposer.expectationsCompleted();
+    observer.newExpectations()
+      .expectRenderSlot('A', 'root')
+      .expectRenderSlot('B', 'item')
+      .expectRenderSlot('C', 'item')
```

Comment:
a bunch of stuff got removed here, are we not going to test the rendering content at all, or shall there still be a todo?

Reply:
My plan is to remove much of the rendering content testing for now, and bring some back later in some tests designed to do exactly that. I'll add this info to a ThingsToDo issue as we discussed.